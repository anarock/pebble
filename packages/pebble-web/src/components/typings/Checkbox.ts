import { RadioProps } from "./Radio";

export type CheckboxProps<OptionType> = RadioProps<OptionType>;
