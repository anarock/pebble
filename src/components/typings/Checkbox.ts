import { RadioProps } from "./Radio";

export type CheckboxProps = RadioProps;
