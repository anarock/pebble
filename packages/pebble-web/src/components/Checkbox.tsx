import * as React from "react";
import Control from "./shared/Control";
import { CheckboxProps } from "./typings/Checkbox";

const Checkbox: React.FunctionComponent<CheckboxProps> = (
  props: CheckboxProps
) => <Control {...props} type="checkbox" />;
export default Checkbox;
