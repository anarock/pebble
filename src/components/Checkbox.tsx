import * as React from "react";
import Control from "./shared/Control";
import { CheckboxProps } from "./typings/Checkbox";

export default function Checkbox<OptionType>(props: CheckboxProps<OptionType>) {
  return <Control {...props} type="checkbox" />;
}
