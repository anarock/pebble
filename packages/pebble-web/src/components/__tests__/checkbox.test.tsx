import * as React from "react";
import Checkbox from "../Checkbox";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from "sinon";

describe("Checkbox", () => {
  test("snapshot", () => {
    const checkbox = renderer.create(
      <Checkbox
        checked
        onChange={() => {}}
        value="checkbox"
        label="I am a checkbox"
      />
    );

    const tree = checkbox.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should call onChange on click with correct arguments", () => {
    const spy = sinon.spy();

    const checkbox = mount(
      <Checkbox onChange={spy} value="checkbox" label="I am a checkbox" />
    );

    checkbox.simulate("click");

    expect(
      spy.calledWith({
        value: "checkbox",
        checked: true
      })
    ).toBeTruthy();
  });

  test("should not call onChange if disabled", () => {
    const spy = sinon.spy();

    const checkbox = mount(
      <Checkbox
        onChange={spy}
        value="checkbox"
        label="I am a checkbox"
        disabled
      />
    );

    checkbox.simulate("click");

    expect(spy.called).toBeFalsy();
  });
});
