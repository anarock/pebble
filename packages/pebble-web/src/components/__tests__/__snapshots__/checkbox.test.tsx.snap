// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox snapshot 1`] = `
.emotion-0 {
  margin-right: 10px;
  font-size: 16px;
  padding-top: 2px;
  height: 18px;
}

.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  padding: 10px 0;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
}

.emotion-1[data-disabled='true'] {
  cursor: not-allowed;
}

.emotion-1[data-disabled='true'] .i {
  color: #E0E0E0;
}

<div
  aria-checked={true}
  className="emotion-1"
  onClick={[Function]}
  role="checkbox"
  tabIndex={0}
>
  <div
    className="emotion-0"
  >
    <i
      className="pi pi-checkbox-selected"
      style={
        Object {
          "color": "#6161FF",
        }
      }
    />
  </div>
   
  I am a checkbox
</div>
`;
